react/wbooks/src/app/components/Routes/components/AuthenticatedRoute.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Redirect, Route, withRouter } from 'react-router-dom';
+import { bool } from 'prop-types';
+
+import { isNull } from '../../../../utils/helpers';
+import { Routes, ACCESS_TOKEN } from '../../../../constants';
+
+const DEFAULT_PUBLIC_ROUTE = Routes.LOGIN_AND_HOME;
+const DEFAULT_PRIVATE_ROUTE = Routes.LOGIN_AND_HOME;
+
+function AuthenticatedRoute({ isPublicRoute, component: Comp, ...props }) {
+  return (
+    <Route
+      {...props}
+      render={routeProps => {
+        const userIsLogged = !isNull(localStorage.getItem(ACCESS_TOKEN));
+        if (isPublicRoute && userIsLogged) {
+          return (
+            <Redirect
+              to={{
+                pathname: DEFAULT_PRIVATE_ROUTE,
+                state: { from: props.location }
+              }}
+            />
+          );
+        } else if (!isPublicRoute && !userIsLogged) {
+          return (
+            <Redirect
+              to={{
+                pathname: DEFAULT_PUBLIC_ROUTE,
+                state: { from: props.location }
+              }}
+            />
+          );
+        }
+
+        return <Comp {...routeProps} />;
+      }}
+    />
+  );
+}
+
+AuthenticatedRoute.defaultProps = {
+  isPublicRoute: false
+};
+
+AuthenticatedRoute.propTypes = {
+  ...Route.propTypes, // eslint-disable-line react/forbid-foreign-prop-types
```

Comment:
this is not a very good way to do it, accessing static properties is not a good practice